react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import ArrowButton from './arrow-button';
+import Image from './image';
+import Dot from './dot';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.shiftLeft = this.shiftLeft.bind(this);
+    this.shiftRight = this.shiftRight.bind(this);
+    this.handleSelect = this.handleSelect.bind(this);
+    this.resetTimeout = this.resetTimeout.bind(this);
+    this.state = {
+      showingIndex: 0,
+      timeoutId: null
+    };
+  }
+
+  shiftLeft(event) {
+    const lastIndex = this.props.images.length - 1;
+    const { showingIndex } = this.state;
+    const nextId = showingIndex === 0 ? lastIndex : showingIndex - 1;
+    const timeoutId = this.resetTimeout();
+    this.setState({ showingIndex: nextId, timeoutId });
+  }
+
+  shiftRight(event) {
+    const lastIndex = this.props.images.length - 1;
+    const { showingIndex } = this.state;
+    const nextId = showingIndex === lastIndex ? 0 : showingIndex + 1;
+    const timeoutId = this.resetTimeout();
+    this.setState({ showingIndex: nextId, timeoutId });
+  }
+
+  handleSelect(event) {
+    const $dot = event.target.closest('.dot');
```

Comment:
In general, we try to avoid traversing the DOM in React. `event.target` is fine, but searching for another element (like `.closest()` or `.querySelector()`) isn't desirable. Fortunately, I don't think it's necessary in your case. Does `event.target.id` contain what you want?